src/OpenTelemetry/Resources/OtelEnvResourceDetector.cs

```diff
@@ -37,9 +39,9 @@ public Resource Detect()
         {
             var resource = Resource.Empty;
 
-            if (this.configuration.TryGetStringValue(EnvVarKey, out string envResourceAttributeValue))
+            if (this.configuration.TryGetStringValue(EnvVarKey, out string? envResourceAttributeValue))
             {
-                var attributes = ParseResourceAttributes(envResourceAttributeValue);
+                var attributes = ParseResourceAttributes(envResourceAttributeValue!);
```

Comment:
Might be a bit of a wild scenario, but seems it _could_ be possible that the configuration providers have been bound in such a way that envResourceAttributeValue is null.

Reply:
Nevermind. `TryGetStringValue` is one of our own extension methods. `envResourceAttributeValue` can't be null.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/f191e846d37283e82272f6daa95df10aec4edd7b/src/OpenTelemetry/Internal/ConfigurationExtensions.cs#L42-L53

Reply:
Thanks for checking! One of the reasons I think it is important to do this nullable decoration is you do find the odd bug here and there when you go through the exercise 😄